src/lib_db.ml

```diff
@@ -144,33 +144,36 @@ let create findlib ~scopes internal_libraries =
 let internal_libs_without_non_installable_optional_ones t =
   String_map.values t.instalable_internal_libs
 
+let interpret_lib_dep t ~dir lib_dep =
+  match lib_dep with
+  | Lib_dep.Direct name -> begin
+      match find_exn t ~from:dir name with
+      | x -> Inl [x]
+      | exception e ->
+        (* Call [find] again to get a proper backtrace *)
+        Inr { fail = fun () -> ignore (find_exn t ~from:dir name : Lib.t); raise e }
```

Comment:
Would you mind explaining this trick a bit? How does calling find_exn again get us a proper back trace? Isn't it going to just raise and we won't re-raise e?

Reply:
yes, that's the expectation. In fact we could just replace the `raise e` by `assert false`. By calling `find_exn`, we get a more precise backtrace. For instance it might start in findlib.ml

Reply:
OK, I think doing `assert false` will be a little nicer since we won't even have to bind `e` in the pattern match.